registrations/views.py

```diff
@@ -104,17 +104,22 @@ def post(self, request, *args, **kwargs):
         return Response(resp, status=status)
 
 
-class RegistrationPost(mixins.CreateModelMixin, generics.GenericAPIView):
+class RegistrationPostPatch(mixins.CreateModelMixin, mixins.UpdateModelMixin,
+                            generics.GenericAPIView):
     permission_classes = (IsAuthenticated,)
     queryset = Registration.objects.all()
     serializer_class = RegistrationSerializer
+    lookup_field = 'id'
 
     def post(self, request, *args, **kwargs):
         # load the users sources - posting users should only have one source
         source = Source.objects.get(user=self.request.user)
         request.data["source"] = source.id
         return self.create(request, *args, **kwargs)
 
+    def patch(self, request, *args, **kwargs):
```

Comment:
The `UpdateModelMixin` gives us the `partial_update` and `update` methods. The `UpdateApiView` class is the one that provides the `put` and `patch` classes that uses these mixins: https://github.com/encode/django-rest-framework/blob/a251b9379200420062cad9e3c68fde7c0e6b3fdc/rest_framework/generics.py#L222-L231